src/plugins/kibana/server/routes/api/settings/register_set.js

```diff
@@ -0,0 +1,33 @@
+import Boom from 'boom';
+import { assign } from 'lodash';
+import defaultsProvider from './defaults';
+
+export default function registerSet(server) {
+  server.route({
+    path: '/api/kibana/settings/{key}',
+    method: 'POST',
+    handler: async function (req, reply) {
+      const key = req.params.key;
+      const value = req.query.value;
```

Comment:
I feel like we should require the value key.  Otherwise `POST /api/kibana/settings/something` is actually a DELETE